src/bin/vip-validate.js

```diff
@@ -0,0 +1,419 @@
+	logToConsole();
+	// If there is a Aborted test result
+	if ( resultCounter[ TestResultType.Aborted ] ) {
+		logToConsole( `${ chalk.bold.bgRedBright( '  NOT PASS  ' ) } There was a critical failure that makes the application ` +
+			'incompatible with VIP Go. Please review the results and re-run the tests.' );
+		process.exit( 1 );
+	}
+
+	// If there is only a partial success, but no failures
+	if ( resultCounter[ TestResultType.PartialSuccess ] && ! resultCounter[ TestResultType.Failed ] ) {
+		logToConsole( `${ chalk.bold.bgYellow( '  PASS  ' ) } The application has passed the required tests, but it does not follow all the recommendations.` );
+		logToConsole( 'Please review the results.' );
+		process.exit( 0 );
+	}
+
+	// If there is a failure
+	if ( resultCounter[ TestResultType.Failed ] ) {
+		logToConsole( `${ chalk.bold.bgRed( '  NOT PASS  ' ) } The application has failed some tests, and will very likely have problems in a production ` +
+			'environment. Please review all the errors found in the results.' );
+		process.exit( 0 );
+	}
+
+	logToConsole( `${ chalk.bold.bgGreen( '  PASS  ' ) } Congratulations. The application passes all the tests.` );
+	process.exit( 0 );
+}
+
+async function validateArgs( opt ): Promise<{}> {
+	const args = {};
+
+	// Verbose
+	if ( opt.verbose ) {
+		Harmonia.setVerbosity( true );
+	}
+
+	// Set path
+	if ( opt.path ) {
+		Harmonia.setCwd( opt.path );
+	}
+
+	// If the JSON option is enabled, all the stdout should be suppressed to prevent polluting the output.
+	if ( opt.json ) {
+		suppressOutput = true;
+		outputJson = true;
+	}
+
+	// Get build information from API and store it in the env object
+	const buildConfig = await getBuildConfiguration( opt.app, opt.env );
+
+	args.nodejsVersion = opt.nodeVersion ?? buildConfig.nodeJSVersion;
+	args.buildType = buildConfig.buildType;
+	args.npmToken = buildConfig.npmToken;
+	args.nodeBuildDockerEnv = buildConfig.nodeBuildDockerEnv;
+
+	args.wait = opt.wait ?? 3000;
+	args.port = opt.port ?? Math.floor( Math.random() * 1000 ) + 3001; // Get a PORT from 3001 and 3999
+
+	return args;
+}
+
+command( {
+	appContext: true,
+	appQuery,
+	envContext: true,
+	module: 'harmonia',
+} )
+	.option( 'verbose', 'Increase logging level to include app build and server boot up messages', false )
+	.option( 'node-version', 'Select a specific target Node.JS version in semver format (MAJOR.MINOR.PATCH) or a MAJOR' )
+	.option( 'wait', 'Configure the time to wait in ms for the app to boot up. Do not change unless you have issues', 3000 )
+	.option( [ 'p', 'port' ], 'Configure the port to use for the app (defaults to a random port between 3001 and 3999)' )
+	.option( 'json', 'Output the results as JSON', false )
+	.option( [ 'P', 'path' ], 'Path to the app to be tested', process.cwd() )
+	.examples( [
+		{
+			usage: 'vip @mysite.production validate',
+			description: 'Run checks and tests and validate your local environment against the production environment',
```

Comment:
"checks and tests"

Is there a solid distinction between these two that the user should care about?

Can we cut it down to either "checks", or "tests", and say that these are used to validate the env?

```suggestion
			description: 'Run checks to validate your local environment against the production environment',
```

Or even make it simpler:

```suggestion
			description: 'Validate your local environment against the production environment',
```

I ran `composer validate --help` to see what it says:

```
The validate command validates a given composer.json and composer.lock

  Exit codes in case of errors are:
  1 validation warning(s), only when --strict is given
  2 validation error(s)
  3 file unreadable or missing

  Read more at https://getcomposer.org/doc/03-cli.md#validate
```

- I think the "The validate command" is redundant, but it says nothing about running checks or tests.
- It describes the error codes - we could have this Harmonia command explain about the success/partial/fail outputs.
- There's a link to documentation.

Can any of this be used as inspiration to improve the inline docs (maybe outside of this `.examples` section)?

Reply:
That is a pretty good suggestion! The only issue is that the current implementation of the VIP-CLI doesn't allow to add these kind of additional help descriptions. The only things that seems possible to add to the sub-command synopsis are command options and examples. 

@chriszarate do you have any suggestions on how this could be included?